src/main/java/org/testng/internal/Parameters.java

```diff
@@ -706,6 +710,25 @@ public static ParameterHolder handleParameters(final ITestNGMethod testMethod,
                                                  final IAnnotationFinder annotationFinder,
                                                  final Object fedInstance,
                                                  final Collection<IDataProviderListener> dataProviderListeners) {
+    return handleParameters(testMethod, allParameterNames, instance, methodParams,xmlSuite, annotationFinder, fedInstance,
+            dataProviderListeners, "@Test");
+  }
+
+  /**
+   * If the method has parameters, fill them in. Either by using a @DataProvider
+   * if any was provided, or by looking up <parameters> in testng.xml
+   * @return An Iterator over the values for each parameter of this
+   * method.
+   */
+  public static ParameterHolder handleParameters(final ITestNGMethod testMethod,
+                                                 final Map<String, String> allParameterNames,
+                                                 final Object instance,
+                                                 final MethodParameters methodParams,
+                                                 final XmlSuite xmlSuite,
+                                                 final IAnnotationFinder annotationFinder,
+                                                 final Object fedInstance,
+                                                 final Collection<IDataProviderListener> dataProviderListeners,
+                                                 final String annotationName) {
```

Comment:
No `final` on parameters (also not a big fan of the alignment since it causes more work when modifying the parameter list, but not a huge deal).

Reply:
Thanks @cbeust That was copy paste error :) The previous method had the `final` keyword too. Cleaned up the final keyword usage in both the places.